ros1_bridge/__init__.py

```diff
@@ -765,12 +823,26 @@ def __init__(self, ros1_msg, ros2_msg):
         self.fields_2_to_1 = OrderedDict()
         self.depends_on_ros2_messages = set()
 
-    def add_field_pair(self, ros1_field, ros2_field):
-        self.fields_1_to_2[ros1_field] = ros2_field
-        self.fields_2_to_1[ros2_field] = ros1_field
-        if ros2_field.type.pkg_name and ros2_field.type.pkg_name != 'builtin_interfaces':
-            self.depends_on_ros2_messages.add(
-                Message(ros2_field.type.pkg_name, ros2_field.type.type))
+    def add_field_pair(self, ros1_fields, ros2_fields):
+        """
+        Add a new mapping for a pair of ROS 1 and ROS 2 messages.
+
+        :type ros1_fields: either a genmsg.msgs.Field object with additional attributes `pkg_name`
+        and `msg_name` as defined by `update_ros1_field_information`, or a tuple of objects of
+        that type
+        :type ros2_field: either a rosidl_adapter.parser.Field object, or a tuple objects of
+        that type
+        """
+        if not type(ros1_fields) is tuple:
```

Comment:
This likely results in a linter warning: should be `not in`.

Same below.

Reply:
Change it to use 'not isinstance(ros1_fields, tuple)'